[Mutes] NoneType error for guild

### What Red version are you using?

3.5.1

### What were you trying to do?

Started bot up, noticed errors in logs from Mutes.

### What did you expect to happen?

No errors to occur.

### What actually happened?

The below error started being spammed
```py
[ERROR] red.cogs.mutes: An unexpected error occurred in the unmute task
Traceback (most recent call last):
May 10 18:32:24 deimos python[116489]:   File "/home/flare/redenv/lib/python3.11/site-packages/redbot/cogs/mutes/mutes.py", line 274, in _clean_tasks
May 10 18:32:24 deimos python[116489]:     r = task.result()
May 10 18:32:24 deimos python[116489]:         ^^^^^^^^^^^^^
May 10 18:32:24 deimos python[116489]:   File "/home/flare/redenv/lib/python3.11/site-packages/redbot/cogs/mutes/mutes.py", line 489, in _auto_channel_unmute_user
May 10 18:32:24 deimos python[116489]:     member = channel.guild.get_member(data["member"])
May 10 18:32:24 deimos python[116489]:              ^^^^^^^^^^^^^
May 10 18:32:24 deimos python[116489]: AttributeError: 'NoneType' object has no attribute 'guild'
```

### How can we reproduce this error?

Unsure of exact steps, seems to be coming from the unmute event that is ran.

### Anything else?

_No response_

--- a/redbot/cogs/mutes/mutes.py
+++ b/redbot/cogs/mutes/mutes.py
@@ -1729,7 +1729,7 @@ class Mutes(VoiceMutes, commands.Cog, metaclass=CompositeMetaClass):
 
         if channel.id not in self._channel_mutes:
             self._channel_mutes[channel.id] = {}
-        current_mute = self._channel_mutes.get(channel.id) or None
+        current_mute = self._channel_mutes[channel.id].get(user.id)
 
         # Determine if this is voice mute -> channel mute upgrade
         is_mute_upgrade = (

--- a/redbot/cogs/mutes/mutes.py
+++ b/redbot/cogs/mutes/mutes.py
@@ -405,8 +405,6 @@ class Mutes(VoiceMutes, commands.Cog, metaclass=CompositeMetaClass):
                             self._unmute_tasks[task_name] = asyncio.create_task(
                                 self._auto_channel_unmute_user(guild_channel, mute_data)
                             )
-                        else:
-                            await self.config.channel_from_id(channel).clear()
 
         del multiple_mutes
 

--- a/redbot/cogs/mutes/mutes.py
+++ b/redbot/cogs/mutes/mutes.py
@@ -401,9 +401,12 @@ class Mutes(VoiceMutes, commands.Cog, metaclass=CompositeMetaClass):
                         log.debug(f"Creating task: {task_name}")
                         if task_name in self._unmute_tasks:
                             continue
-                        self._unmute_tasks[task_name] = asyncio.create_task(
-                            self._auto_channel_unmute_user(guild.get_channel(channel), mute_data)
-                        )
+                        if guild_channel := guild.get_channel(channel):
+                            self._unmute_tasks[task_name] = asyncio.create_task(
+                                self._auto_channel_unmute_user(guild_channel, mute_data)
+                            )
+                        else:
+                            await self.config.channel_from_id(channel).clear()
 
         del multiple_mutes
 
@@ -1726,7 +1729,7 @@ class Mutes(VoiceMutes, commands.Cog, metaclass=CompositeMetaClass):
 
         if channel.id not in self._channel_mutes:
             self._channel_mutes[channel.id] = {}
-        current_mute = self._channel_mutes.get(channel.id)
+        current_mute = self._channel_mutes.get(channel.id) or None
 
         # Determine if this is voice mute -> channel mute upgrade
         is_mute_upgrade = (

